thrift/thrift0.13/src/test/java/com/linecorp/armeria/server/thrift/THttpServiceBlockingTest.java

```diff
@@ -0,0 +1,127 @@
+
+    @BeforeEach
+    void clearDetector() {
+        blocking.set(false);
+    }
+
+    @AfterAll
+    public static void shutdownExecutor() {
+        executor.shutdown();
+    }
+
+    @RegisterExtension
+    static final ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+
+            sb.service("/",
+                       THttpService.builder().addService(new HelloServiceAsyncImpl()).build());
+            sb.service("/blocking", THttpService.builder()
+                                                .useBlockingTaskExecutor(true)
+                                                .addService(new HelloServiceAsyncImpl())
+                                                .build());
+            sb.service("/blocking-iface",
+                       THttpService.builder().addService(new HelloServiceImpl()).build());
+
+            sb.blockingTaskExecutor(executor, true);
+        }
+    };
+
+    @Test
+    void nonBlocking() throws Exception {
+        final HelloService.Iface client = ThriftClients.newClient(server.httpUri(), HelloService.Iface.class);
+
+        final String message = "nonBlockingTest";
+        final String response = client.hello(message);
+
+        assertThat(response).isEqualTo(message);
+        assertThat(blocking).isFalse();
+    }
+
+    @Test
+    void blocking() throws Exception {
+        final HelloService.Iface client =
+                ThriftClients.builder(server.httpUri())
+                             .path("/blocking")
+                             .build(HelloService.Iface.class);
+        final String message = "blockingTest";
+        final String response = client.hello(message);
+
+        assertThat(response).isEqualTo(message);
+        assertThat(blocking).isTrue();
+    }
+
+    @Test
+    void blockingIface() throws Exception {
+        final HelloService.Iface client =
+                ThriftClients.builder(server.httpUri())
+                             .path("/blocking-iface")
+                             .build(HelloService.Iface.class);
+        final String message = "blockingTest";
+        final String response = client.hello(message);
+
+        assertThat(response).isEqualTo(message);
+        assertThat(blocking).isTrue();
+    }
+
+    static class HelloServiceAsyncImpl implements HelloService.AsyncIface {
+        @Override
+        public void hello(String name, AsyncMethodCallback resultHandler) throws TException {
+            resultHandler.onComplete(name);
+            blocking.set(Thread.currentThread().getName().startsWith(BLOCKING_EXECUTOR_PREFIX));
+        }
+    }
+
+    static class HelloServiceImpl implements HelloService.Iface {
+        @Override
+        public String hello(String name) throws TException {
+            blocking.set(Thread.currentThread().getName().startsWith(BLOCKING_EXECUTOR_PREFIX));
```

Comment:
Optional; setting the thread name itself instead of a boolean may help with debugging issues.

e.g. we can differentiate between 1) was a thread set at all vs. 2) was the wrong thread set

Reply:
Done!